framework/web/Application.php

```diff
@@ -118,6 +188,64 @@ public function handleRequest($request)
 
     private $_homeUrl;
 
+    /**
+     * @param string $route
+     * @return array|bool
+     */
+    public function createController($route)
+    {
+        $parts = parent::createController($route);
+        if (is_array($parts)) {
+            [$controller, $action] = $parts;
+            if (method_exists($controller, 'middleware')) {
+                foreach ($controller->middleware() as $middleware) {
+                    $this->addMiddleware($middleware);
+                }
+            }
+            $this->callMiddlewareStack($action);
+        }
+        return $parts;
+    }
+
+    /**
+     * Sort middleware by priority and execute
+     *
+     * @param $action string|null
+     * @throws MiddlewareException
+     */
+    protected function callMiddlewareStack($action)
+    {
+        usort($this->middlewareStack, function ($a, $b) {
+            if ($a['priority'] == $b['priority']) {
+                return 0;
+            }
+            return $a['priority'] > $b['priority'] ? -1 : 1;
+        });
+
+        foreach ($this->middlewareStack as $middleware) {
+
+            if (in_array($action, (array)$middleware['except'])) {
+                continue;
+            }
+            if (count($middleware['only'])>0 && !in_array($action, $middleware['only'])) {
```

Comment:
Will be a notice in case `only` is not set.
isset + is_array before counting is required

Reply:
Could you add `true` as a 3-rd argument of `in_array`?

Reply:
"Will be a notice in case only is not set.". 
No. All keys always exists. See https://github.com/yiisoft/yii2/pull/15294/files#diff-d21ce18380a9785b6cee4643f50873f1R97